MessageCenter/Classes/Views/Chatting/GroupChannelChattingViewController.swift

```diff
@@ -631,6 +629,33 @@ class GroupChannelChattingViewController: UIViewController, SBDConnectionDelegat
         self.present(mediaUI, animated: true, completion: nil)
     }
     
+    @objc private func sendFileMessage() {
+        let status = PHPhotoLibrary.authorizationStatus()
+        if status == .authorized {
+            openPicker()
+        } else {
+            PHPhotoLibrary.requestAuthorization { status in
+                switch status {
+                case .authorized:
+                    self.openPicker()
+                    break
+                case .denied, .restricted:
+                    DispatchQueue.main.async {
+                        let vc = UIAlertController(title: "Error", message: "Authorization to assets is denied.", preferredStyle: UIAlertControllerStyle.alert)
+                        let closeAction = UIAlertAction(title: "Close", style: UIAlertActionStyle.cancel, handler: nil)
+                        vc.addAction(closeAction)
+                        DispatchQueue.main.async {
```

Comment:
No need to run on DispatchQueue.main since you're already in a main thread block